transformer_lens/HookedTransformer.py

```diff
@@ -133,9 +139,20 @@ def __init__(
             self.init_weights()
 
         if move_to_device:
-            # Move the model to the relevant device, suppress the logging message
-            self.to(self.cfg.device, print_details=False)
-        
+            # We load the devices in a pipeline manner - the first device gets the embed and pos_embed layers and the first n_layers // n_devices blocks,
+            # the second gets the next n_layers // n_devices blocks ... the last gets the last n_layers // n_devices blocks, the final
+            # normalization layer (if it exists) and the unembed layer
+            self.embed.to(utils.get_device_for_block_index(0, self.cfg))
+            self.hook_embed.to(utils.get_device_for_block_index(0, self.cfg))
+            if self.cfg.positional_embedding_type != "rotary":
+                self.pos_embed.to(utils.get_device_for_block_index(0, self.cfg))
+                self.hook_pos_embed.to(utils.get_device_for_block_index(0, self.cfg))
+            if hasattr(self, "ln_final"):
+                self.ln_final.to(utils.get_device_for_block_index(self.cfg.n_layers - 1, self.cfg))
+            self.unembed.to(utils.get_device_for_block_index(self.cfg.n_layers - 1, self.cfg))
+            for i, block in enumerate(self.blocks):
+                block.to(utils.get_device_for_block_index(i, self.cfg))
```

Comment:
When testing this, I suspect you'll want to move all of this into its own class method on `HookedTransformer` as well (but up to you - just depends on how you test it). I think a key ways it could break is if we (a) forget to move one of the modules to a device or (b) move a module to an incorrect device with e.g. an off-by-one error.

This could also keep the init method a little more manageable (it's huge).